src/joy.c

```diff
@@ -1021,21 +847,23 @@ int process_directory_of_files (char *input_directory, char *output_filename) {
     memset_s(&fp,  sizeof(struct bpf_program), 0x00, sizeof(struct bpf_program));
     memset_s(&pcap_filename[0], (MAX_FILENAME_LEN*2), 0x00, (MAX_FILENAME_LEN*2));
 
-    /* create a directory to place all of the output files into */
+    /* use the output filename as the directory to storing results */
     if (glb_config->filename) {
-        struct stat st;
+        strncpy_s(output_dir, (MAX_FILENAME_LEN-1), glb_config->filename, (MAX_FILENAME_LEN-1));
+
+        /* create a directory to place all of the output files into */
         memset_s(&st,  sizeof(struct stat), 0x00, sizeof(struct stat));
-        if (stat(output_filename, &st) == -1) {
+        if (stat(output_dir, &st) == -1) {
 #ifdef WIN32
-            mkdir(output_filename);
+            mkdir(output_dir);
 #else
-            tmp_ret = mkdir(output_filename, 0700);
-	    if (tmp_ret < 0) {
-		joy_log_err("Error creating directory: %s\n", output_filename);
-		return tmp_ret;
-	    }
+            tmp_ret = mkdir(output_dir, 0700);
+            if (tmp_ret < 0) {
+                joy_log_err("Error creating directory: %s\n", output_dir);
```

Comment:
Do we want to print the system error here. strerror(errno)

Reply:
not required